lib/open3.rb

```
@@ -445,6 +461,9 @@ def popen2(*cmd, &block)
   # in the call to Process.spawn;
   # see {Execution Options}[https://docs.ruby-lang.org/en/master/Process.html#module-Process-label-Execution+Options].
   #
+  # Note: The redirection options :in and [:out, :err] will be ignored
```

Q: `:in and [:out, :err]` is it meant to be different from how the other methods are documented?

---

I thought that was a little weird, too.
It must be because `[:out, :err]` is actually what's used in the code of that method:
https://github.com/ruby/open3/pull/24/files#diff-22509663fa03dbb4fc6fa062734ed1eed523f8d125e604da6e303d8102036bceR535-R539
So if you specified that key exactly, it will be silently ignored, which was the theme of these updates.

I just tested what happens if you pass `:err` (or `:out`) specifically, then you get an error that the FD was specified twice.
We probably don't need to document that `[:out, :err]` is specifically silently ignored because that seems like an implementation detail that could be changed.
Maybe instead we should change the language to say something more like "just don't specify any of the redirection handles"?

I suppose we could also change the code to warn about them being passed (and ignored) (so that it isn't silent)?

What do you think?

---

I made the message more general, what do you think about this?